### PR TITLE
Metrics

### LINKED ISSUE #1774: Istio's Prow should use the new upstream monitoring and alerting stack.
https://github.com/kubernetes/test-infra/tree/master/prow/cluster/monitoring#monitoring
We'll get lots of nice preconfigured graphs and alerts for free. e.g. https://monitoring.prow.k8s.io
I think we can pretty easily move over anything from https://velodrome.istio.io that is still used too so that we can turn down velodrome and just have a single monitoring dashboard.

cc @clarketm @fejta @geeknoid @howardjohn

### PR DESCRIPTION
**Pull request by scottilee**
Add files required for a custom monitoring and alerting stack planned for monitoring.prow.istio.io.

- Sends alerts to the Istio Slack channel #test-alerts

Fixes #1774.

### CODE REVIEW

File: prow/cluster/monitoring/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,100 @@
+load("//prow:def.bzl", "BUILD_CLUSTER", "MULTI_KIND", "component", "release")
```
Comment (clarketm) [inline prow/cluster/monitoring/BUILD.bazel]: Istio is not setup for `bazel` deployments. We can safely remove this (and the other) `BUILD.bazel` files from this PR. As an alternative, we can use `kubectl` (w/ `helm` for magic strings) but that can be in a follow-up.

File: prow/cluster/monitoring/OWNERS [hunk 2]
```diff
@@ -0,0 +1,3 @@
+approvers:
```
Comment (clarketm) [inline prow/cluster/monitoring/OWNERS]: You can omit this `OWNERS` file as ownership is already handled by [`CODEOWNERS`](https://github.com/istio/test-infra/blob/master/CODEOWNERS).

**Review by clarketm — CHANGES_REQUESTED** [review]

File: prow/cluster/monitoring/OWNERS (hunk 2 above)
Comment (scottilee) [reply prow/cluster/monitoring/OWNERS]: Done.

File: prow/cluster/monitoring/BUILD.bazel (hunk 1 above)
Comment (scottilee) [reply prow/cluster/monitoring/BUILD.bazel]: Done.

**Comment by clarketm** [comment]
@scottilee – regarding the lint errors, can you omit the `vendor` directory and update your PR? 

Managing these vendor dependencies and auto-deploying can be a follow-up item.

**Review by clarketm — APPROVED** [review]
This is awesome, thanks for the help! I will create follow-up tasks for updating the README, installing vendor deps, and setting up auto-deployment.

**Pull request merged**